napari/layers/points/points.py

```diff
@@ -1479,8 +1479,8 @@ def selected_data(self) -> Selection[int]:
 
     @selected_data.setter
     def selected_data(self, selected_data: Sequence[int]) -> None:
-        self._selected_data.clear()
-        self._selected_data.update(set(selected_data))
+        self._selected_data.intersection_update(selected_data)
+        self._selected_data.update(selected_data)
```

Comment:
Would be nice to add a comment here to quickly explain why these 2 lines! Other than that, looks good to me :+1:

Reply:
Sorry for the delay, I have added an explanatory comment